### ISSUE TITLE
*obj* doesn't implement the cuda array interface. at cuda.as_cuda_array(D.detach())

### ISSUE THREAD

**Issue by KimUyen** [Maghoumi/pytorch-softdtw-cuda#1]
Dear Maghoumi,
Firstly, I would like to thank your effort on this implementation.

I am trying to use your code, but i am getting an error: **\*obj\* doesn't implement the cuda array interface.**
At the code block: 
``` 
# Run the CUDA kernel.
# Set CUDA's grid size to be equal to the batch size (every CUDA block processes one sample pair)
# Set the CUDA block size to be equal to the length of the longer sequence (equal to the size of the largest diagonal)
compute_softdtw_cuda[B, threads_per_block](cuda.as_cuda_array(D.detach()),
                                        gamma.item(), bandwidth.item(), N, M, n_passes,
                                        cuda.as_cuda_array(R))
```
Could you please help me know where the reason is?
_My configuration:
GPU: GeForce RTX 2080 Ti
Cuda: Cuda compilation tools, release 10.0, V10.0.130
Cuda driver: version 452.06
torch: version 1.6.0_

**Comment by KimUyen** [comment]
Thanks Maghoumi,
I tried to pass all to CUDA and it worked now. Thanks for your support.
